docs/src/openapi-v1.yaml

```
@@ -273,6 +273,56 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Error'
+  /blocks/{blockId}/extrinsics/{extrinsicsIndex}:
+    get:
+      tags:
+      - blocks
+      summary: Get an extrinsic by its extrinsicIndex and block height or hash.
```

```suggestion
      summary: Get an extrinsic by its `Timepoint` (blockId, extrinsicIndex).
```
@danforbes @joepetrowski thoughts?

---

Is "Timepoint" and established term in our documentation? If not I think it sounds a bit funny. I found the [`pallet_multisig::Timepoint`](https://substrate.dev/rustdocs/v2.0.0/pallet_multisig/struct.Timepoint.html) but not sure how widely used it is.

---

Yea `Timepoint` does sound a little funny, but that being said a lot of terminology can be weird sometimes, so if this is the correct verbiage I agree with it.

---

Yeah, only usage is in Multisig. I think we can also use this to unify the concept. "Get an extrinsic by its `extrinsicIndex` and block height or hash. The pair `blockId, extrinsicIndex` is sometimes referred to as a `Timepoint`."
https://github.com/paritytech/substrate/blob/d1923a4bc1eded37e57062f8b501c646e5321221/frame/multisig/src/lib.rs#L105

---

@joepetrowski So would you think keeping the same name `getExtrinsicsByExtrsinsicIndex` for the new API route is the way to go, or changing it too `getExtrinsicsByTimepoint`?

---

I'm warming up to `Timepoint` but if we go that route we should be prepared to shill it across the board.

---

Yup exactly. I totally agree, we would definitely have to make it completely consistent across the board.

---

Yeah, I'm on board with `getExtrinsicByTimepoint`. (Note the _singular_ `Extrinsic`. You either get all _extrinsics_ from a block, but a `Timepoint` always refers to a single extrinsic.)